### PR TITLE
Added some meat to the bone :D

### PR DESCRIPTION
**Pull request by CodyPredy**
Added repo app with client and node support
Added support for styled components
Some updates to docker for node debugging

### CODE REVIEW

File: client/src/components/Panel.js [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+  }
+
+  .content {
+    text-align: left;
+    padding: 15px;
+    font-size: 14px;
+  }
+`;
+
+export default Panel;
```
Comment (intojs) [inline client/src/components/Panel.js]: I know the AirBnB style guide favours defaults. I hate them :) It promotes inconsistency when renaming stuff. But. do not change, it's valid code :)

File: client/src/components/PanelContainer.js [hunk 2]
```diff
@@ -0,0 +1,29 @@
+import React from "react";
+import styled from "styled-components";
+
+import Panel from "./Panel";
+
+function PanelContainer({ content = [] }) {
+  return !content.length ? (
```
[the remark this answers is not in the document]
Comment (intojs) [reply client/src/components/PanelContainer.js]: Also, in ternaries, think of using:

```
a > b
   ? ...
   : ....
```

You can see the two conditions better.

File: client/src/components/PanelContainer.js [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+    <NoContent>Repo man is not here</NoContent>
+  ) : (
+    <Container>
+      {content.map((panelProps, index) => (
```
Comment (intojs) [inline client/src/components/PanelContainer.js]: Maybe hit an enter between the curly and content?

File: client/src/components/PanelContainer.js [hunk 4]
```diff
@@ -0,0 +1,29 @@
+import React from "react";
+import styled from "styled-components";
+
+import Panel from "./Panel";
+
+function PanelContainer({ content = [] }) {
```
[the remark this answers is not in the document]
Comment (intojs) [reply client/src/components/PanelContainer.js]: Maybe rename content to panels?

File: client/src/components/PanelContainer.js (hunk 3 above)
Comment (intojs) [reply client/src/components/PanelContainer.js]: If content becomes panels, you can rename panelProps to panel

File: client/src/components/PanelContainer.js [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+  return !content.length ? (
+    <NoContent>Repo man is not here</NoContent>
+  ) : (
+    <Container>
+      {content.map((panelProps, index) => (
+        <Panel key={`${panelProps.name}${index}`} {...panelProps} />
```
Comment (intojs) [inline client/src/components/PanelContainer.js]: 💯

File: client/src/components/PanelContainer.js [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+      ))}
+    </Container>
+  );
+}
+
+const Container = styled.div`
```
Comment (intojs) [inline client/src/components/PanelContainer.js]: I think it's just luck and how the code gets transpiled, in declaring Components after usage.

File: client/src/index.css [hunk 7]
```diff
@@ -1,8 +1,8 @@
 body {
   margin: 0;
   padding: 0;
-  font-family: -apple-system, BlinkMacSystemFont, "Segoe UI", "Roboto", "Oxygen",
-    "Ubuntu", "Cantarell", "Fira Sans", "Droid Sans", "Helvetica Neue",
+  font-family: "Roboto", -apple-system, BlinkMacSystemFont, "Segoe UI", "Roboto",
```
Comment (intojs) [inline client/src/index.css]: Why so many font families?

File: client/src/views/RepoViews.js [hunk 8]
```diff
@@ -0,0 +1,84 @@
+import React, { useState } from "react";
+import { http } from "../api/http";
+
```
Comment (intojs) [inline client/src/views/RepoViews.js]: Remove this line please

File: client/src/views/RepoViews.js [hunk 9]
```diff
@@ -0,0 +1,84 @@
...
+  function renderUserInput() {
+    return (
+      <RepoInput>
+        <input
+          value={author}
+          onChange={e => {
+            setAuthor(e.target.value);
+          }}
+          onKeyPress={e => onAuthorSubmit(e)}
```
Comment (intojs) [inline client/src/views/RepoViews.js]: You can just use onAuthorSubmit

File: server/src/api/getRepos.js [hunk 10]
```diff
@@ -0,0 +1,31 @@
+const axios = require("axios");
+
+module.exports = async function getRepos(user) {
+  try {
+    const { data } = await axios.get(
+      `https://api.github.com/users/${user}/repos`
+    );
+    const repos = data.map(
+      ({
+        name = "",
```
Comment (intojs) [inline server/src/api/getRepos.js]: 💯

File: server/src/index.js [hunk 11]
```diff
@@ -14,11 +14,9 @@ server.use((req, res, next) => {
...
 });
 
-server.get("/api", (req, res) => {
-  fs.readFile("./src/api/response.json", (err, json) => {
-    let obj = JSON.parse(json);
-    res.json(obj);
-  });
+server.get("/api/:name", async (req, res) => {
+  const response = await getRepos(req.params.name);
+  res.json(response);
```
Comment (intojs) [inline server/src/index.js]: Please handle the error case

File: server/src/index.js [hunk 12]
```diff
@@ -1,5 +1,5 @@
 const express = require("express");
-const fs = require("fs");
+const getRepos = require("./api/getRepos");
```
Comment (intojs) [inline server/src/index.js]: Try to separate the libraries from own packages by grouping them together and splitting the code with an empty line.

File: client/src/App.js [hunk 13]
```diff
@@ -1,27 +1,14 @@
 import React, { Component } from "react";
-import logo from "./logo.svg";
 import "./App.css";
 
-import Hello from "./components/hello";
+import RepoViews from "./views/RepoViews";
```
Comment (intojs) [inline client/src/App.js]: I feel like, even if it's not subject to this change, that the css should be imported first maybe? Usually, when I modify a component, I almost never modify the css file, but I do add other imports. And then, the order of imports breaks.

File: client/src/components/Panel.js [hunk 14]
```diff
@@ -0,0 +1,78 @@
...
+      </div>
+    </PanelContent>
+  );
+}
+
+const PanelContent = styled.div`
+  width: 45%;
+  display: flex;
+  justify-content: flex-start;
+  box-sizing: border-box;
```
Comment (intojs) [inline client/src/components/Panel.js]: You should order all the css alphabetically :)

Comment (CodyPredy) [reply client/src/components/Panel.js]: Shhh... We don't want to be in the minority:
https://css-tricks.com/poll-results-how-do-you-order-your-css-properties/

File: client/src/components/Panel.js [hunk 15]
```diff
@@ -0,0 +1,78 @@
...
+        <h4>
+          <a href={clone}>{name}</a>
+        </h4>
+        <p>{description}</p>
+      </div>
+    </PanelContent>
+  );
+}
+
+const PanelContent = styled.div`
```
[the remark this answers is not in the document]
Comment (CodyPredy) [reply client/src/components/Panel.js]: Well either we add them first or we add them last. 
The majority of cases I've seen is to add them after the fact.

Why you may ask?
Well the majority of the time the imports, class or component definitions are clustered to the top of the code. Add styling there and things can get pretty cluttered.

I prefer to add them last to keep styling separate and less obtrusive(optimally I would still keep them in a separate file but that's a bigger hassle).

File: client/src/components/PanelContainer.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (CodyPredy) [reply client/src/components/PanelContainer.js]: Valid. Got carried away when debugging.

File: client/src/views/RepoViews.js [hunk 16]
```diff
@@ -0,0 +1,84 @@
...
+          GO
+        </button>
+      </RepoInput>
+    );
+  }
+
+  return (
+    <div>
+      {renderUserInput()}
+      {isLoading ? (
```
Comment (robert-damoc) [inline client/src/views/RepoViews.js]: I think that here you can use the same style as @intojs said above:
```
a > b
   ? ...
   : ....
```

File: client/src/views/RepoViews.js [hunk 17]
```diff
@@ -0,0 +1,84 @@
...
+        <img src={logo} className="App-logo" alt="logo" />
+      ) : (
+        <PanelContainer content={content} />
+      )}
+    </div>
+  );
+}
+
+const RepoInput = styled.div`
```
Comment (robert-damoc) [inline client/src/views/RepoViews.js]: I'm curious, why are we using styles like this instead of having them in a `.(s)css` file?

Comment (CodyPredy) [reply client/src/views/RepoViews.js]: https://blog.bitsrc.io/5-ways-to-style-react-components-in-2019-30f1ccc2b5b

Comment (CodyPredy) [reply client/src/views/RepoViews.js]: Here is an interesting read for you:
https://blog.bitsrc.io/5-ways-to-style-react-components-in-2019-30f1ccc2b5b

File: client/src/views/RepoViews.js (hunk 16 above)
Comment (CodyPredy) [reply client/src/views/RepoViews.js]: Neah... I agree with using negations but the indentation not so much. React makes ternaries pretty visible in the case:
```
{isLoading ? (
    <img src={logo} className="App-logo" alt="logo" />
) : (
    <PanelContainer content={content} />
)}
```

File: client/src/App.js (hunk 13 above)
Comment (CodyPredy) [reply client/src/App.js]: Updated.

**Pull request merged**